### ISSUE TITLE
Support configuring the className of the <input> element

### ISSUE THREAD

**Issue by tobia** [Hacker0x01/react-datepicker#105]
Thank you for this great component!

For those using a CSS library (eg. Bootstrap) there is usually a specific class that we need to add to all `<input>` elements, such as `form-control`, for them to follow the library style.

Right now we can replace `datepicker__input` with `form-control` in the datepicker JS file (what I'm doing) or copy all the Bootstrap CSS with the datepicker class name.

It would be cool to have `className` on the Datepicker passed to the input element.

**Comment by RSO** [comment]
Interesting suggestion. I always imagined that _at some point_ we needed some sort of varaible classNames. If I understand you correctly you're suggesting to only add the option to change the input className?

**Comment by tobia** [comment]
I used your datepicker on a Bootstrap-based website, and yes, the input was the only className I had to tweak.

I made the following change to my copy of the JS file:

	-	      className: "datepicker__input",
	+	      className: "form-control ignore-react-onclickoutside",

`form-control` is Bootstrap's class that has to be applied to form inputs to give them the library's look.

I also added `ignore-react-onclickoutside` to allow users to choose a date with the picker and then tweak it in the text input, or vice versa. Without it, the picker would close whenever the user clicked on the input, which I thought was sub-optimal.
